src/main/java/net/io/kino/service/impl/AuthenticationManagerImpl.java

```diff
@@ -0,0 +1,68 @@
+package net.io.kino.service.impl;
+
+import net.io.kino.repository.ManagersRepository;
+import net.io.kino.model.Manager;
+import net.io.kino.service.AuthenticationManager;
+import org.apache.tomcat.util.codec.binary.Base64;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import javax.crypto.SecretKey;
+import javax.crypto.SecretKeyFactory;
+import javax.crypto.spec.PBEKeySpec;
+import java.security.NoSuchAlgorithmException;
+import java.security.SecureRandom;
+import java.security.spec.InvalidKeySpecException;
+
+public class AuthenticationManagerImpl implements AuthenticationManager {
+
+    private static final int iterations = 20*1000;
+    private static final int saltLen = 32;
+    private static final int desiredKeyLen = 256;
+
+    @Autowired
+    private ManagersRepository managers;
+
+    @Override
+    public boolean login(String username, String password) throws InvalidKeySpecException, NoSuchAlgorithmException {
+        return authenticate(username, password);
+    }
+
+    @Override
+    public boolean logout(Manager user) {
+        return false;

```

Comment:
Podejrzewam, że ta metoda będzie zmieniona, ale jako sama deklaracja to co oznacza zwracany boolean z logout? Może się wylogowywanie nie udać? Jak się nie uda to co mogłoby to znaczyć i co z tym możemy zrobić?

Reply:
Tam wpisałam w komentarzu, że to do zmian. Pomysł był taki, żeby raczej po prostu dać sygnał, że wylogowywanie się powiodło, gdyby któryś z modułów tego potrzebował (mięliśmy prezentacje przed frontendem, także woleliśmy to uwzględnić). Myślę, że jednak zmienię to na void, ale jeszcze muszę pomyśleć, jak ta metoda powinna wyglądać.